src/libraries/System.Private.CoreLib/src/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -1559,20 +1558,22 @@ private void EnterSpin(EnterSpinLockReason reason)
                     Interlocked.Add(ref _enterDeprioritizationState, deprioritizationStateChange);
                 }
 
-                int processorCount = Environment.ProcessorCount;
-                for (int spinIndex = 0; ; spinIndex++)
+                for (uint spinIndex = 0; ; spinIndex++)
                 {
-                    if (spinIndex < LockSpinCount && processorCount > 1)
+                    // - Spin-wait until the Sleep(0) threshold
+                    // - Beyond the Sleep(0) threshold, alternate between sleeping and spinning. Avoid using only Sleep(0), as
```

Comment:
> Avoid using only Sleep(0), as it may be ineffective when there are no other threads waiting to run.

On a single processor, this will just use Thread.Sleep(0)... is that ok?

Reply:
On a single processor that's what is done elsewhere, it's unlikely that a meaningful change would occur while spin-waiting without yielding.